docs/plugins/workspace.md

```diff
@@ -0,0 +1,171 @@
+about access control).
+
+A new attribute, `workspaces`, is being added to saved objects which type is an array of string. A saved object can be associated with one
+or multiple workspaces. The saved objects(dashboards, visualizations, etc) will only be displayed in the associated workspaces.
+
+The follow example shows the dashboard object is associated with workspace which id is `M5NqCu`
+```typescript
+{
+  type: "dashboard",
+  id: "da123f20-6680-11ee-93fa-df944ec23359",
+  workspaces: ["M5NqCu"]
+}
+```
+
+Saved object can also associated with multiple workspaces, this is useful in scenarios where a saved object is relevant
+to multiple teams, projects, or use cases.
+
+Consider the following example, where a visualization is associated with multiple workspaces:
+```typescript
+{
+  type: "visualization",
+  id: "da123f20-6680-11ee-93fa-df944ec23359",
+  workspaces: ["M5NqCu", "<TeamA-workspace-id>", "<Analytics-workspace-id>"]
+}
+```
+By allowing saved objects to be linked with multiple workspaces, this enables users to share and collaborate on resources
+across various workspaces(teams).
+
+## Copy saved objects among workspaces
+While associating saved objects with multiple workspaces links a single object instance to multiple places, copying saved 
+objects takes a different approach. When copying objects, it creates hard copies of the objects in the target workspace,
+regardless of their original workspaces.
+
+For example, if copy this object to `<target-workspace>`
+```typescript
+{
+  type: "visualization",
+  id: "da123f20-6680-11ee-93fa-df944ec23359",
+  workspaces: ["M5NqCu", "<TeamA-workspace-id>", "<Analytics-workspace-id>"]
+}
+```
+Then a new object will be created with new `id` and associated with `<target-workspace>`
+```typescript
+{
+  type: "visualization",
+  id: "<new-object-id>",
+  workspaces: ["<target-workspace>"]
+}
+```
+### Handling Dependencies
+A significant aspect of copying saved objects is the handling of dependencies. Many saved objects, particularly
+visual objects like dashboards and visualizations, often have a hierarchical structure with dependencies.
+For example, a dashboard may depend on multiple visualizations, and each visualization may rely on specific
+index pattern objects.
+
+The copying process is not limited to the saved object itself. Instead, it copies the entire dependency hierarchy
+associated with the saved object. For example:
+1. If the visualization depends on specific index pattern objects, these index pattern objects will also be duplicated
+    in `<target-workspace>`.
+2. If the dashboard depends on multiple visualizations, those visualizations and their associated index patterns will
+    be copied as well.
+
+This ensures that the copied saved object in <target-workspace> retains its functionality and context, with all necessary
```

Comment:
There is a duplicate related object flag that's selected by default, but user can also choose not to duplicate related object.

Reply:
Updated on line 136